source/common/stats/symbol_table_impl.h

```diff
@@ -465,6 +363,56 @@ class StatNameTempStorage : public StatNameStorage {
   SymbolTable& symbol_table_;
 };
 
+// Represents an ordered container of StatNames. The encoding for each StatName
+// is byte-packed together, so this carries less overhead than allocating the
+// storage separately. The tradeoff is there is no random access; you can only
+// iterate through the StatNames.
+//
+// The maximum size of the list is 255 elements, so the length can fit in a
+// byte. It would not be difficult to increase this, but there does not appear
+// to be a current need.
```

Comment:
As an aside, I keep thinking that we shouldn't have to reinvent the byte packed design here, I'm sure this is done elsewhere for efficient string packing, but I don't see anything in STD or absl..

Reply:
Understood; I'm sure this has been implemented somewhere, notably utf-8 encoders, but if it isn't in an already-used library it seems like it's easier to just have a robust tested impl here.  WDYT?

Reply:
I'd also be willing to factor this out and make it a separate tested packed-int abstraction in source/common/common, but I'd prefer to do that in a separate PR, as that code is already stable in the system and this PR is just abstrating the API to it.